### PR TITLE
Unbreak xine-lib, xine-ui and dvdstyler

### PR DESCRIPTION
**Pull request by c0bw3b**
###### Motivation for this change

`xine-lib` is currently broken on master since the latest libcaca bump. And it will also break with ffmpeg 5.x which is currently on staging.
This reuses patches from Arch for both problems.

`dvdstyler` is broken on master and on 21.11 because it crashes on startup because of a symbols conflict between GTK2 and GTK3. It works only with wxGTK30 without GTK2.
It was also missing wrapping for Gnome apps.
(I will later backport the second commit)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by c0bw3b** [comment]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>dvdstyler</li>
    <li>eaglemode</li>
    <li>quodlibet-xine</li>
    <li>quodlibet-xine-full</li>
    <li>xine-lib</li>
    <li>xine-ui</li>
  </ul>
</details>

**Comment by AndersonTorres** [comment]
Why does it fail in ARM Linux?

**Comment by c0bw3b** [comment]
It is `xine-ui` failing, not `xine-lib`.
And [it's failing on Hydra](https://hydra.nixos.org/job/nixpkgs/trunk/xine-ui.aarch64-linux) since it was bumped from 0.99.12 to 0.99.13

**Comment by AndersonTorres** [comment]
Ok then.

File: pkgs/development/libraries/xine-lib/default.nix [hunk 1]
```diff
@@ -33,7 +35,21 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-71GyHRDdoQRfp9cRvZFxz9rwpaKHQjO88W/98o7AcAU=";
   };
 
+  patches = [
+    (fetchpatch {
```
Comment (AndersonTorres) [inline pkgs/development/libraries/xine-lib/default.nix:39]: Will these patches be incorporated in the future?

Comment (c0bw3b) [reply pkgs/development/libraries/xine-lib/default.nix:39]: I just checked upstream Mercurial repo and they do have those changes in their tree.

libcaca fix is https://sourceforge.net/p/xine/xine-lib-1.2/ci/3f383e140270cd45448bd9ebaa08232579e1c1af/

ffmpeg fix is https://sourceforge.net/p/xine/xine-lib-1.2/ci/4e5ad7ebed6111ba9e1ea2eab4a6b94477b066a8/

Comment (AndersonTorres) [reply pkgs/development/libraries/xine-lib/default.nix:39]: As I have seen here, sourceforge has no native way to [download patch files](https://sourceforge.net/p/forge/feature-requests/614/).

Just to make sure, and [to make hammer happy](https://github.com/jtojnar/nixpkgs-hammering), comment why these patches are needed, plus an advice to remove them at the next release.

Comment (c0bw3b) [reply pkgs/development/libraries/xine-lib/default.nix:40]: Done

**Comment by c0bw3b** [comment]
We would need [this change](https://sourceforge.net/p/xine/xine-ui/ci/f9a98e02460348ca23ca2f5c2b39e62c8758b22d/) to fix `xine-ui` on aarch64.
I'll see if someone already made a patch of that, or add it in our tree.

**Comment by c0bw3b** [comment]
@ofborg build xine-ui

File: pkgs/development/libraries/xine-lib/default.nix [hunk 2]
```diff
@@ -33,7 +35,23 @@ stdenv.mkDerivation rec {
...
+    # Fix build with ffmpeg 5.0 ; remove for xine-lib 1.2.12
+    (fetchpatch {
+      name = "xine-lib-ffmpeg-5.0-fix.patch";
+      url = "https://raw.githubusercontent.com/archlinux/svntogit-community/209ae10d59d29c13633b75aa327cf937f3ff0725/trunk/020-xine-lib-ffmpeg-5.0-fix.patch";
+      sha256 = "15ff15bqxq1nqqazfbmfq6swrdjr2raxyq7hx6k0r61izhf0g8ld";
+    })
+  ];
+
   nativeBuildInputs = [
+    autoconf
```
Comment (AndersonTorres) [inline pkgs/development/libraries/xine-lib/default.nix:54]: Remember myself to try autoreconfHook here

**Review by AndersonTorres — APPROVED** [review]

**Pull request merged**